### ISSUE TITLE
imkafka topic support   wildcard

### ISSUE THREAD

**Issue by zhxiaom5** [rsyslog/rsyslog#1643]
imkafka is  a wondful plugin,but this plugin doesnt support wildcard topic.
add support seems to be a good idea

**Comment by rgerhards** [comment]
can you elaborate a bit?

**Comment by jay7x** [comment]
I'm sure it's about having wildcards in imkafka 'topic' parameter. Like in Graylog you may specify regex to match topics to consume from. E.g. when you have 'rsyslog.messages' and 'rsyslog.auth' topics you may consume from both using 'rsyslog\..*' regex.

**Comment by rgerhards** [comment]
@alorbach would like to have a look?

**Comment by zhxiaom5** [comment]
@jay7x 
yes,thats what i mean. if imkafka  topic config  support wildcards  match or regex  match,that would be great.
thanks

**Comment by rgerhards** [comment]
Sorry for the very late follow-up. This answer may no longer be relevant to the original reporter, but I am adding it for people arriving through search while we work through the backlog. We do not lightly close issues, and this is not being closed merely because of its age.

Current `imkafka` uses librdkafka consumer subscriptions (`rd_kafka_subscribe`). In that API, a topic entry beginning with `^` is treated as a regular-expression subscription, so this request is already supported. For the example discussed above, configure the topic as a regex such as:

```rsyslog
input(type="imkafka"
      topic="^rsyslog\\..*"
      broker=["broker:9092"]
      consumergroup="rsyslog")
```

This is a regex subscription rather than shell glob syntax; the leading `^` is what selects regex matching. It also follows newly created matching topics through librdkafka metadata updates.

I am closing this as already implemented. If this does not work with a current rsyslog and librdkafka version, please open a new issue with that version, configuration, and startup diagnostics so we can treat it as a current bug.
